framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts

```diff
@@ -177,6 +187,10 @@ export class ReportMisbehaviorPlugin extends BasePlugin {
 							transaction: encodedTransaction,
 						});
 
+						// Set new currentHeight
+						if (decodedBlockHeader.height > this._state.currentHeight) {
```

Comment:
so this will be executed only when `contradictingBlock` is true.. a scenario where upto x million blocks, no contradicting blocks found the `this._state.currentHeight` will be zero? so the `clearBlockHeaders` job will not clear anything?
- from height 0 to 3million blocks not contradiction found so, `this._state.currentHeight` will be zero?
- when `this._clearBlockHeadersInterval` time is triggered the `this._state.currentHeight` will be zero so nothing happens?